CONTRIBUTING.md

```diff
@@ -0,0 +1,662 @@
+
+```
+«scope»: «summary»
+
+«motivation for change»
+```
+
+Where the scope is one of:
+
+| Scope          | Purpose                                                          |
+|----------------|------------------------------------------------------------------|
+| `ci`           | Changes to GitHub Actions workflows.                             |
+| `doc`          | Changes to the website, `README.md`, and so on.                  |
+| `flake`        | Changes in the `flake` directory, `flake.nix`, and `flake.lock`. |
+| `stylix`       | Changes in the `stylix` directory and other global code.         |
+| Name of target | Changes to code for a particular target.                         |
+| `treewide`     | Changes across many targets.                                     |
+
+The scope is meant to indicate which area of the code was changed. Specifying
+the type of change, such as `feat` or `fix`, is not necessary. Dependency
+updates should use whichever scope they are related to.
+
+The summary should start with a lowercase letter, and should not end with
+punctuation.
+
+Most commits to `master` will also include a pull request number in brackets
+after the summary. GitHub adds this automatically when creating a squash merge.
+
+## Adding modules
+
+### Development setup
+
+Currently the easiest way to test Stylix is to use the new code in your actual
+configuration.
+
+You might find it useful to override Stylix' input flake reference on your
+flake, from `github:nix-community/stylix` to
+`git+file:/home/user/path/to/stylix`, so that you don't need to push changes to
+GitHub during testing.
+
+To do that, instead of editing your `flake.nix`, you can leverage `nix`'
+`--override-input` parameter (which can also be supplied through their
+frontends: `nixos-rebuild`, `nix-on-droid` and even `nh`). It allows you to
+deploy your changes in one fell swoop, without having to update the lock file of
+your flake every time you make an edit.
+
+Just append `--override-input stylix git+file:/home/user/path/to/stylix` to your
+standard `nix` (or `nix` frontend) incantation.
+
+Nix only reads files which are tracked by Git, so you also need to `git add
+«file»` after creating a new file.
+
+### Module naming
+
+Modules should be named like `modules/«name»/«platform».nix`. For example,
+`modules/avizo/hm.nix` is a Home Manager module which themes Avizo.
+
+The following platforms are supported:
+
+- NixOS (`nixos`)
+- Home Manager (`hm`)
+- Nix-Darwin (`darwin`)
+- Nix-on-Droid (`droid`)
+
+Correctly named modules will be imported automatically.
+
+Other files needed by the module can also be stored within the `modules/«name»`
+folder, using any name which is not on the list above.
+
+### Module template
+
+Modules should be created using the `mkTarget` function whenever possible (see
+the [`/stylix/mk-target.nix`](
+https://github.com/danth/stylix/blob/-/stylix/mk-target.nix) in-source
```

Comment:
In nixvim, we would write this as a repo-relative link:


```suggestion
the [`/stylix/mk-target.nix`](
./stylix/mk-target.nix) in-source
```

And our fix-links wrapper would auto-convert it to an external link when copying the file into the docs drv.

Reply:
This isn't really important, but it'd have some subtle benefits:

In forks, the link would stay within the fork.

In the docs, you could link to the specific commit; like you already do for other links to stylix's repo.

The owner would automatically be in sync with other repo links; this is still using the old repo owner.